File: evidence/generated-systems/golang/healthcare-on-golang/healthcare-on-golang/internal/model/Laboratory.go
package model

import (
    "gorm.io/gorm"
)

//==============================================================
// Laboratory Declaration
//==============================================================
type Laboratory struct {
    gorm.Model
     Name                                    string
    CliaNumber                                    string
    FacilityId         *uint
    Facility           *Facility `gorm:"foreignKey:FacilityId"`
     LaboratoryOrders           []LaboratoryOrder `gorm:"foreignKey:LaboratoryOrdersFromLaboratoryId"`
     LabResults           []LabResult `gorm:"foreignKey:LabResultsFromLaboratoryId"`

// parent associations as their child

}

File: evidence/generated-systems/django/healthcare-on-django/healthcareOnDjango/models/Laboratory.py
from django.db import models

#======================================================================
# 
# Encapsulates data for model Laboratory
#
# @author Harbormaster Dev Team
#
#======================================================================

#======================================================================
# Class Laboratory Declaration
#======================================================================
class Laboratory (models.Model):

#======================================================================
# attribute declarations
#======================================================================
	name = models.CharField(max_length=200, null=True)
	cliaNumber = models.CharField(max_length=200, null=True)
	facility = models.ForeignKey('Facility', on_delete=models.CASCADE, null=True, blank=True, related_name='+')
	laboratoryOrders = models.ManyToManyField('LaboratoryOrder',  blank=True, related_name='+')
	labResults = models.ManyToManyField('LabResult',  blank=True, related_name='+')

#======================================================================
# function declarations
#======================================================================
	def toString(self):
		str = ""
		str = str + self.name
		str = str + self.cliaNumber
		return str;
    
	def __str__(self):
		return self.toString();

	def identity(self):
		return "Laboratory";
    
	def objectType(self):
		return "Laboratory";
